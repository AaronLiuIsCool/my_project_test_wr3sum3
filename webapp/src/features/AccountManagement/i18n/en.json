{
    "heading": "账号管理",
    "basicInfo": "基本信息",
    "accountManagement": "账号管理",
    "wechat": "微信",
    "email": "邮箱",
    "password": "密码",
    "bind": "前往绑定",
    "unbind": "解除绑定"
  }