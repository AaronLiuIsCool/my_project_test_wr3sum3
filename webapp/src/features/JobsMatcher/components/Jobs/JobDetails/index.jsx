import React from 'react';

import DetailHeader from './DetailHeader';
import DetailContent from './DetailContent';

import styles from '../../../styles/JobDetails.module.css';

const renderDetail = (data) => (
    <>
        <DetailHeader data={data} />
        <DetailContent data={data} />
    </>
);

const JobDetails = ({ data }) => (
    <div className={styles["container-details"]}>
        {Boolean(data) ? renderDetail(data) : null}
    </div>
);

export default JobDetails;