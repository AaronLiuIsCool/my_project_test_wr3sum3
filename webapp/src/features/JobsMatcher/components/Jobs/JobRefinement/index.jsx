import React, { useState } from 'react';
import JROverWriteResume from './JROverWriteResume';
import JRRenameResume from './JRRenameResume';
import JRResumerDetails from './JRResumerDetails';

import styles from '../../../styles/JobRefinement.module.css';
import CloseHoverIcon from '../../../assets/close_hover.svg';
import CloseRegularIcon from '../../../assets/close_regular.svg';

const renderModalSteppers = (data, modalOpenHandler, step, setStep) => {


	return (
		<>
			<div className='closeIconContainer' style={{ top: '-40px', right: '10px' }} onClick={() => modalOpenHandler(false)}>
				<img src={CloseRegularIcon} alt='Close' className='closeIcon' />
				<img src={CloseHoverIcon} alt='Close' className='closeIconHover' />
			</div>
			{step === 1 && <JROverWriteResume stepHandler={setStep} />}
			{step === 2 && <JRRenameResume stepHandler={setStep} />}
			{step === 3 && <JRResumerDetails stepHandler={setStep} data={data} />}
		</>
	);
};

const JobRefinement = ({ data, modalOpenHandler }) => {
	const [step, setStep] = useState(1); 
	return (
		<div className={styles.container}>
			<div className={styles.body}>{Boolean(data) ? renderModalSteppers(data, modalOpenHandler, step, setStep) : null}</div>
		</div>
	);
};

export default JobRefinement;
