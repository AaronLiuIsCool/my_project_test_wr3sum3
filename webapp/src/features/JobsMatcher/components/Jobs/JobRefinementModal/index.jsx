import React, { useState } from 'react';
import { useHistory, Redirect } from 'react-router-dom';
import JROverWriteResume from './JROverWriteResume';
import JRRenameResume from './JRRenameResume';

import styles from '../../../styles/JobRefinementModal.module.css';
import CloseHoverIcon from '../../../assets/close_hover.svg';
import CloseRegularIcon from '../../../assets/close_regular.svg';

const renderModalSteppers = (jobId, resumeId, modalOpenHandler, step, setStep) => {

	return (
		<>
			<div className='closeIconContainer' style={{ top: '-40px', right: '10px' }} onClick={() => modalOpenHandler(false)}>
				<img src={CloseRegularIcon} alt='Close' className='closeIcon' />
				<img src={CloseHoverIcon} alt='Close' className='closeIconHover' />
			</div>
			{step === 1 && <JROverWriteResume stepHandler={setStep} />}
			{step === 2 && <JRRenameResume />}
			{step === 3 && <Redirect push to={`/refinement?jobId=${jobId}&resumeId=${resumeId}`} />}
		</>
	);
};

const JobRefinementModal = ({ data, modalOpenHandler, selectedJob }) => {
	const history = useHistory();

	const [step, setStep] = useState(1); 
	const [resumeId] = useState(history.location.search.slice(8)); 

	return (
		<div className={styles.container}>
			<div className={styles.body}>{Boolean(data) ? renderModalSteppers(data.jobs[selectedJob].jobUuid, resumeId, modalOpenHandler, step, setStep) : null}</div>
		</div>
	);
};

export default JobRefinementModal;
