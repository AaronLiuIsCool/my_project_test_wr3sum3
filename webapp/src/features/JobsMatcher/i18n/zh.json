{
  "search_query_label": "工作岗位搜索",
  "search_query_placeholder": "输入职位或公司",
  "search_country_label": "国家",
  "search_city_label": "城市",
  "search_button": "搜索",

  "job_list_summary": "找到{0}份和您所填经历匹配的工作",
  "job-details-save": "收藏",
  "job-details-referral": "联系内推",
  "job-details-apply": "申请工作", 
  "job_matching_score": "简历匹配度：还行",
  "job_finishing_touches": "对岗精修",
  "job_details_description": "职位描述"
}