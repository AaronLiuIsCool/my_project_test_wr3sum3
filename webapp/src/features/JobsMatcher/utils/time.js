export function timeSince(dateStr) {
  const date = new Date(dateStr);
  const seconds = Math.floor((new Date() - date) / 1000);
  let interval = seconds / 31536000;

  if (interval > 1) {
      return `${Math.floor(interval)}年前发布`;
  }
  interval = seconds / 2592000;
  if (interval > 1) {
      return `${Math.floor(interval)}月前发布`;
  }
  interval = seconds / 86400;
  if (interval > 1) {
      return `${Math.floor(interval)}天前发布`;
  }
  interval = seconds / 3600;
  if (interval > 1) {
      return `${Math.floor(interval)}小时前发布`;
  }
  interval = seconds / 60;
  if (interval > 1) {
      return `${Math.floor(interval)}分钟前发布`;
  }
  return `${Math.floor(interval)}秒前发布`;
}
