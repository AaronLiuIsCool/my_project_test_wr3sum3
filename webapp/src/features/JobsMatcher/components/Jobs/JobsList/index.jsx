import React from 'react';

import Pagination from 'react-bootstrap/Pagination'
import Job from './Job';

import { useI8n } from 'shell/i18n';

import styles from '../../../styles/JobsList.module.css';

const TOTAL_PAGINATION_ITEMS = 4;

export function calculate(pageNumber, maxPages) {
    const shouldShowPrev = pageNumber > 0;
    const shouldShowNext = pageNumber < maxPages;
    let min = Math.max(0, pageNumber - 1);
    let max = Math.min(maxPages, pageNumber + 2);

    if ((max - min + 1) < TOTAL_PAGINATION_ITEMS && (min !== 0 || max !== maxPages)) {
        if (min === 0) {
            max = min + TOTAL_PAGINATION_ITEMS - 1;
        } else {
            min = max - TOTAL_PAGINATION_ITEMS + 1;
        }
    }
    return {shouldShowPrev, shouldShowNext, min, max};
}

const getPaginationItems = (pageNumber, maxPages, onPageChange) => {
    const {shouldShowPrev, shouldShowNext, min, max} = calculate(pageNumber, maxPages);
    const items = [];
    for (let i = min; i <= max; i++) {
        items.push(
            <Pagination.Item active={pageNumber === i} onClick={() => onPageChange(i)} key={`pageItem-${i}`}>
                {i+1}
            </Pagination.Item>
        );
    }

    return (
        <Pagination>
            {shouldShowPrev ? <Pagination.Prev onClick={() => onPageChange(pageNumber-1)}/> : null}
            {items}
            {shouldShowNext ? <Pagination.Next onClick={() => onPageChange(pageNumber+1)}/> : null}
        </Pagination>
    );
}

const JobsList = ({ data, pageNumber, onPageChange, selection, onSelect }) => {
    const messages = useI8n();

    return (
        <div className={styles["container-list"]}>
            <span className={styles["list-summary"]}>
                {messages["job_list_summary"].replace('{0}', data.total || 0)}
            </span>
            {data && data.content && data.content.map((job, index) => (
                <Job data={job} key={`job-${index}`}
                    onClick={() => onSelect(index)}
                    selected={selection === index} />
            ))}
            {getPaginationItems(pageNumber, Math.floor(data.total/data.pageSize), onPageChange)}
        </div>
    );
}

export default JobsList;