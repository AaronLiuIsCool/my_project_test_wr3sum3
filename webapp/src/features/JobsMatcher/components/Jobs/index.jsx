import React, { useState } from 'react';

import JobsList from './JobsList';
import JobDetails from './JobDetails';

import styles from '../../styles/Jobs.module.css';

const Jobs = ({ data, pageNumber, onPageChange }) => {
    const [selection, setSelection] = useState(0);

    return (
        <div className={styles.container}>
            <JobsList data={data} pageNumber={pageNumber} onPageChange={onPageChange}
                onSelect={setSelection} selection={selection} />
            <JobDetails data={data && data.content && data.content[selection]} />
        </div>
    )
};

export default Jobs;