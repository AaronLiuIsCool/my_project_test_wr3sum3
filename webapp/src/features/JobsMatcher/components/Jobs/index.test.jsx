import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from 'store';
import { I8nContext } from 'shell/i18n';

import Jobs from './index';
import * as time from '../../utils/time';

const JobsWithProvider = (props) => (
    <Provider store={store}>
        <I8nContext.Provider value={{
            'job_list_summary': '{0}'
        }}>
            <Jobs {...props} />
        </I8nContext.Provider>
    </Provider>
);

const mockData = {
    "jobs":[
        {
            "jobUuid":"4dd748d4-0b22-11eb-9fa5-a56d63ee61ee",
            "title":"Software Development Engineer",
            "companyName":"微软",
            "location":{
                "country":"中国",
                "state":"",
                "city":"北京"
            },
            "jobType":"Full-time",
            "relevantMajors":[
                "Computer Science"
            ],
            "url":"https://cn.linkedin.com/jobs/view/software-development-engineer-at-%E5%BE%AE%E8%BD%AF-1998181365?utm_campaign=google_jobs_apply&utm_source=google_jobs_apply&utm_medium=organic",
            "postDate":"2020-10-04T17:59:12.768"
        },
        {
            "jobUuid":"7a8a44c2-1180-11eb-8f88-916881325b13",
            "title":"Software Engineer (University Hiring)",
            "companyName":"HPE",
            "location":{
                "country":"中国",
                "state":"Beijing",
                "city":"北京"
            },
            "url":"https://cn.jobrapido.com/jobpreview/96839405?utm_campaign=google_jobs_apply&utm_source=google_jobs_apply&utm_medium=organic",
            "postDate":"2020-10-13T20:28:27.43"
        },
        {
            "jobUuid":"4e67ef9a-0b22-11eb-9fa5-1979a4f02776",
            "title":"Analog Design Engineer",
            "companyName":"安森美半导体",
            "location":{
                "country":"中国",
                "state":"",
                "city":"北京"
            },
            "url":"https://cn.linkedin.com/jobs/view/analog-design-engineer-at-%E5%AE%89%E6%A3%AE%E7%BE%8E%E5%8D%8A%E5%AF%BC%E4%BD%93-2020144244?utm_campaign=google_jobs_apply&utm_source=google_jobs_apply&utm_medium=organic",
            "postDate":"2020-10-05T17:59:13.716"
        },
        {
            "jobUuid":"31c40d0a-1180-11eb-8f88-398cde6016b2",
            "title":"AI/ML - Software Engineer Intern, Machine Intelligence",
            "companyName":"Apple Inc.",
            "location":{
                "country":"中国",
                "state":"Beijing",
                "city":"北京"
            },
            "url":"https://cn.jobrapido.com/jobpreview/96834204?utm_campaign=google_jobs_apply&utm_source=google_jobs_apply&utm_medium=organic",
            "postDate":"2020-10-14T20:26:25.335"
        }
    ],
    "offset":0,
    "limit":7,
    "total":42
};

describe('Jobs', () => {
    beforeEach(() => {
        time.timeSince = jest.fn(() => '3 days ago');
    });

    test('render Jobs', async () => {
        const searchHeader = renderer.create(
            <JobsWithProvider data={{}} resumeId={'test id'} />
        );
        expect(searchHeader.toJSON()).toMatchSnapshot();
    });

    test('render Jobs with full data', async () => {
        const searchHeader = renderer.create(
            <JobsWithProvider data={mockData} resumeId={'test id'} />
        );
        expect(searchHeader.toJSON()).toMatchSnapshot();
    });
});
