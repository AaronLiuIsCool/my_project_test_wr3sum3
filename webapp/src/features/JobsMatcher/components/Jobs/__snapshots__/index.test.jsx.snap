// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs render Jobs 1`] = `
<div
  className="container"
>
  <div
    className="container-list"
  >
    <span
      className="list-summary"
    >
      0
    </span>
    <ul
      className="pagination"
    />
  </div>
  <div
    className="container-details"
  />
</div>
`;

exports[`Jobs render Jobs with full data 1`] = `
<div
  className="container"
>
  <div
    className="container-list"
  >
    <span
      className="list-summary"
    >
      42
    </span>
    <div
      className="container-job-selected"
      onClick={[Function]}
    >
      <div
        className="job-title"
      >
        Software Development Engineer
      </div>
      <div
        className="job-company"
      >
        微软
      </div>
      <div
        className="job-post-time"
      >
        3 days ago(stub time since)
      </div>
    </div>
    <div
      className="container-job"
      onClick={[Function]}
    >
      <div
        className="job-title"
      >
        Software Engineer (University Hiring)
      </div>
      <div
        className="job-company"
      >
        HPE
      </div>
      <div
        className="job-post-time"
      >
        3 days ago(stub time since)
      </div>
    </div>
    <div
      className="container-job"
      onClick={[Function]}
    >
      <div
        className="job-title"
      >
        Analog Design Engineer
      </div>
      <div
        className="job-company"
      >
        安森美半导体
      </div>
      <div
        className="job-post-time"
      >
        3 days ago(stub time since)
      </div>
    </div>
    <div
      className="container-job"
      onClick={[Function]}
    >
      <div
        className="job-title"
      >
        AI/ML - Software Engineer Intern, Machine Intelligence
      </div>
      <div
        className="job-company"
      >
        Apple Inc.
      </div>
      <div
        className="job-post-time"
      >
        3 days ago(stub time since)
      </div>
    </div>
    <ul
      className="pagination"
    >
      <li
        className="page-item active"
      >
        <span
          className="page-link"
          disabled={false}
          onClick={[Function]}
        >
          1
          <span
            className="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          2
        </a>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          3
        </a>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          4
        </a>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <span
            aria-hidden="true"
          >
            ›
          </span>
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </li>
    </ul>
  </div>
  <div
    className="container-details"
  />
</div>
`;
